LoopKit/Extensions/DoseEntry+LocalizedDescription.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  DoseEntry+LocalizedDescription.swift
+//  LoopKit
+//
+//  Created by Nathaniel Hamming on 2020-10-27.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+
+extension DoseEntry {
+    static var units = HKUnit.internationalUnit()
+
+    private var numberFormatter: NumberFormatter {
+        let numberFormatter = NumberFormatter()
+        numberFormatter.maximumFractionDigits = DoseEntry.unitsPerHour.maxFractionDigits
+        return numberFormatter
+    }
+
+    public var localizedAttributedDescription: NSAttributedString {
+        let typeAndUnitAttributes: [NSAttributedString.Key: Any] = [
+            .font: UIFont.preferredFont(forTextStyle: .body),
+            .foregroundColor: UIColor.secondaryLabel
+        ]
+
+        switch type {
+        case .basal, .bolus, .tempBasal:
+            let unitString = type == .bolus ? DoseEntry.units.shortLocalizedUnitString() : DoseEntry.unitsPerHour.shortLocalizedUnitString()
+            let value: Double = type == .bolus ? (deliveredUnits ?? programmedUnits) : unitsPerHour
+            
+            let valueAttributes: [NSAttributedString.Key: Any] = [
+                .font: UIFont.preferredFont(forTextStyle: .headline),
+                .foregroundColor: UIColor.label
+            ]
+             
+            let typeAttributedString = NSAttributedString(string: type.localizedDescription + ": ", attributes: typeAndUnitAttributes)
```

Comment:
I think all of this needs to go in a localized string to cover left-to-right ordering and such.